File: scratch/read_excel.js
import XLSX from 'xlsx';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const excelPath = path.resolve(__dirname, '../../hasilAnalisis.xlsx');
const workbook = XLSX.readFile(excelPath);
console.log("Sheet Names:", workbook.SheetNames);

workbook.SheetNames.forEach((sheetName) => {
    const sheet = workbook.Sheets[sheetName];
    console.log(`\n--- Sheet: ${sheetName} (ref: ${sheet['!ref']}) ---`);
    
    const json = XLSX.utils.sheet_to_json(sheet, { header: 1 });
    console.log("First 10 rows:");
    json.slice(0, 10).forEach((row, i) => {
        console.log(`Row ${i + 1}:`, row.slice(0, 8));
    });
});

File: scratch/read_excel.py
import openpyxl
import os

wb = openpyxl.load_workbook("hasilAnalisis.xlsx", data_only=True)
print("Sheet Names:", wb.sheetnames)

for sheetname in wb.sheetnames:
    sheet = wb[sheetname]
    print(f"\n--- Sheet: {sheetname} ({sheet.max_row} rows x {sheet.max_column} cols) ---")
    for r in range(1, min(10, sheet.max_row + 1)):
        row_vals = [sheet.cell(r, c).value for c in range(1, min(8, sheet.max_column + 1))]
        print(f"Row {r}: {row_vals}")
